pyciemss/interfaces.py

```diff
@@ -528,16 +533,204 @@ def wrapped_model():
     return {"inferred_parameters": inferred_parameters, "loss": loss}
 
 
-# # TODO
-# def optimize(
-#     model: CompiledDynamics,
-#     objective_function: ObjectiveFunction,
-#     constraints: Constraints,
-#     optimization_algorithm: OptimizationAlgorithm,
-#     *args,
-#     **kwargs
-# ) -> OptimizationResult:
-#     """
-#     Optimize the objective function subject to the constraints.
-#     """
-#     raise NotImplementedError
+def optimize(
+    model_path_or_json: Union[str, Dict],
+    end_time: float,
+    logging_step_size: float,
+    qoi: Callable,
+    risk_bound: float,
+    static_parameter_interventions: Dict[torch.Tensor, str],
+    objfun: Callable,
+    initial_guess_interventions: List[float],
+    bounds_interventions: List[List[float]],
+    *,
+    solver_method: str = "dopri5",
+    solver_options: Dict[str, Any] = {},
+    start_time: float = 0.0,
+    inferred_parameters: Optional[pyro.nn.PyroModule] = None,
+    n_samples_ouu: int = int(1e2),
+    maxiter: int = 5,
+    maxfeval: int = 25,
+    verbose: bool = False,
+    roundup_decimal: int = 4,
+    postprocess: bool = False,
+) -> Dict:
+    """
+    Load a model from a file, compile it into a probabilistic program, and optimize under uncertainty
+    with risk-based constraints over dynamical models.
+    Args:
+        model_path_or_json: Union[str, Dict]
+            - A path to a AMR model file or JSON containing a model in AMR form.
+        end_time: float
+            - The end time of the sampled simulation.
+        logging_step_size: float
+            - The step size to use for logging the trajectory.
+        qoi: Callable
+            - A callable function defining the quantity of interest to optimize over.
+        risk_bounds: float
+            - The threshold on the risk constraint.
+        static_parameter_interventions: Dict[float, Dict[str, Intervention]]
+            - A dictionary of static interventions to apply to the model.
+            - Each key is the time at which the intervention is applied.
+            - Each value is a dictionary of the form {parameter_name: intervention_assignment}.
```

Comment:
Looks like this was copied from the earlier documentation. I believe it should be edited to say "each value is a string".

Reply:
Fixed